dora/underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -140,6 +146,21 @@ protected HdfsUnderFileSystem(AlluxioURI ufsUri, UnderFileSystemConfiguration co
       Configuration hdfsConf, boolean useLoadingCache) {
     super(ufsUri, conf);
 
+    mTrashEnable = alluxio.conf.Configuration.getBoolean(
+        PropertyKey.UNDERFS_HDFS_TRASH_ENABLE);
+    LOG.info(PropertyKey.UNDERFS_HDFS_TRASH_ENABLE.getName() + " is set to {}", mTrashEnable);
+    mFsTrash = CacheBuilder.newBuilder().build(new CacheLoader<FileSystem, Optional<Trash>>() {
+      @Override
+      public Optional<Trash> load(FileSystem fs) throws Exception {
+        Configuration configuration = fs.getConf();
+        long interval = configuration.getLong(CommonConfigurationKeys.FS_TRASH_INTERVAL_KEY, 0);
+        if (interval == 0) {
+          return Optional.empty();
```

Comment:
Are we essentially reimplementing the switch that when `FS_TRASH_INTERVAL_KEY` is `0`, the trash is disabled? This is however already handled in `TrashPolicyDefault` [here](https://github.com/apache/hadoop/blob/271b4b25cd04663edc043ad47a101b5f4238ab2c/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicyDefault.java#L104-L106). Checking HDFS configuration in Alluxio seems unnecessary and confusing.

Reply:
if FS_TRASH_INTERVAL_KEY == 0, the HDFS `trash` is disabled in UFS.
So, we will not try move the file to trash for each file to be deleted.
This will avoid performance degradation, I think.

@humengyu2012 may explain more on this topic.

Reply:
When the trash(interval <=0) is not enabled, forcefully using it will result in Alluxio being unable to delete files. We have two options:

1. Revert to the regular deletion logic.
2. Report an error to the user.

 In this case, I have chosen option 1.

Reply:
> This will avoid performance degradation

When trash is disabled in HDFS, the `moveToTrash` method returns immediately. So no perf pitfall here. https://github.com/apache/hadoop/blob/a3b9c37a397ad4188041dd80621bdeefc46885f2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicyDefault.java#L124-L125

> When the trash is not enabled(interval <=0), forcefully using it will result in Alluxio being unable to delete files. 

[`‎Trash::moveToTrash‎` returns `false` if the item is already in the trash or trash is disabled](https://github.com/apache/hadoop/blob/a3b9c37a397ad4188041dd80621bdeefc46885f2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Trash.java#L107). How about checking its return value, and fall back to regular deletion if false?

My point here is that you do not have to reimplement the logic of enabling and disabling HDFS trash in Alluxio, which is already handled by HDFS itself.

Reply:
Good idea, we can do like this:
```
if(!moveToTrash(path)){
  delete(path);
}
```
However, I'm concerned that if the moveToTrash method returns false due to other exceptional circumstances, it could result in the data being deleted and irrecoverable.

Reply:
I think we should consider a different approach for implementation: If the user activates the trash feature, we should always use the trash functionality to delete files. If the trash operation returns false, we can directly raise an exception to the user, informing them that there might be a misconfiguration related to the trash settings. This way, we can avoid performing any checks within Alluxio.